pep-0708.rst

```diff
@@ -0,0 +1,846 @@
+PEP: 708
+Title: Preventing Dependency Confusion Attacks
+Author: Donald Stufft <donald@stufft.io>
+PEP-Delegate: Paul Moore <p.f.moore@gmail.com>
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Content-Type: text/x-rst
+Created: 20-Feb-2023
+
+
+Abstract
+========
+
+There is a long-standing class of attacks that are called "dependency confusion"
+attacks, which roughly boil down to an individual user expected to get package
+``A``, but instead they got ``B``. In Python, this almost always happens due to
+the configuration of multiple repositories (possibly including the default of
+PyPI), where they expected package ``A`` to come from repository ``X``, but
+someone is able to publish to repository ``Y`` under the same name.
+
+A specific example of this is the recent case where the PyTorch project had an
+internal project named ``torchtriton`` which was only ever intended to be
+installed from their repositories located at ``https://download.pytorch.org/``,
+but that repository was designed to be used in conjunction with PyPI, and
+the name of ``torchtriton`` was not claimed on PyPI, which allowed the attacker
+to use that name and publish a malicious version.
+
+This PEP proposes an extension to the Repository API that would enable
+installers to determine when it is safe to install a given package that appears
+in multiple repositories, allowing them to be stricter in the cases where they
+do not know if it is safe.
+
+
+Rationale
+=========
+
+Dependency Confusion attacks have long been possible, but they've recently
+gained press with public examples of cases where these attacks were successfully
+executed.
```

Comment:
Linking to some of these examples (or the press regarding them) would seem prudent here.

Reply:
Added some links and moved the `torchtriton` example to here to better support this claim.